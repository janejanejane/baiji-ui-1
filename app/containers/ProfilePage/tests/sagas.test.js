/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { put } from 'redux-saga/effects';
// import { defaultSaga } from '../sagas';

import {
  loadProfileSaga,
} from '../sagas';

import {
  loadProfileSuccess,
  loadProfileFailed,
} from '../actions';

describe('loadProfile Saga', () => {
  let loadProfileGenerator;

  beforeEach(() => {
    loadProfileGenerator = loadProfileSaga({
      id: 'something',
    });

    const testCall = loadProfileGenerator.next().value;
    expect(testCall).toMatchSnapshot();
  });

  describe('OnSuccess', () => {
    it('should dispatch loadProfile action on success', () => {
      const fixture = {
        username: 'testusername',
      };

      const callDescriptor = loadProfileGenerator.next(fixture).value;
      expect(callDescriptor).toEqual(put(loadProfileSuccess(fixture)));
    });
  });

  describe('OnFailed', () => {
    it('should dispatch loadProfile failed on failure', () => {
      const fixture = {
        message: 'Error message',
      };

      const callDescriptor = loadProfileGenerator.throw(fixture).value;
      expect(callDescriptor).toEqual(put(loadProfileFailed(fixture.message)));
    });
  });
});
