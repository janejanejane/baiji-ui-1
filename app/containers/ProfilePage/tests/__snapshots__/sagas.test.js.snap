// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadProfile Saga OnFailed should dispatch loadProfile failed on failure 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/profile?id=something",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`loadProfile Saga OnSuccess should dispatch loadProfile action on success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/profile?id=something",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
